### PR TITLE
wezterm: unstable-2020-11-22 -> 20210314 (+ fix build)

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 1]
```diff
@@ -77,21 +79,17 @@ rustPlatform.buildRustPackage {
 
   buildInputs = runtimeDeps;
 
-  installPhase = "" + lib.optionalString stdenv.isLinux ''
+  preFixup = "" + lib.optionalString stdenv.isLinux ''
     for artifact in wezterm wezterm-gui wezterm-mux-server strip-ansi-escapes; do
-      patchelf --set-rpath "${lib.makeLibraryPath runtimeDeps}" $releaseDir/$artifact
-      install -D $releaseDir/$artifact -t $out/bin
+      patchelf --set-rpath "${lib.makeLibraryPath runtimeDeps}" $out/bin/$artifact
```
Comment (Mic92) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:84]: releaseDir was no longer set, so I changed it to `preFixup`.

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 2]
```diff
@@ -77,21 +79,17 @@ rustPlatform.buildRustPackage {
...
-  cp $releaseDir/wezterm "$OUT_APP"
-  cp $releaseDir/wezterm-mux-server "$OUT_APP"
-  cp $releaseDir/wezterm-gui "$OUT_APP"
-  cp $releaseDir/strip-ansi-escapes "$OUT_APP"
+    mkdir -p "$out/Applications"
+    OUT_APP="$out/Applications/WezTerm.app"
+    cp -r assets/macos/WezTerm.app "$OUT_APP"
+    rm $OUT_APP/*.dylib
+    cp -r assets/shell-integration/* "$OUT_APP"
+    ln -s $out/bin/{wezterm,wezterm-mux-server,wezterm-gui,strip-ansi-escapes} "$OUT_APP"
```
Comment (Mic92) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:92]: It would be great if someone could test this part.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>wezterm</li>
</ul>
</details>

File: pkgs/applications/terminal-emulators/wezterm/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:92]:
```
results/wezterm/bin/wezterm
 2021-04-01T11:59:05.989Z INFO  wezterm_mux_server_impl::local > setting up /run/user/1000/wezterm/gui-sock-374667
 2021-04-01T11:59:05.990Z ERROR wezterm_gui                    > failed to open X11 display; terminating
 2021-04-01T11:59:05.991Z ERROR wezterm_toast_notification::dbus > while showing notification: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files
 ```

I am to lazy to test this not headless but it looks already not quite right.

Comment (Mic92) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:92]: If this is a headless machine without x11 than this looks good to me. I think using dbus on macOS is an upstream bug. There is no dbus-based notifications on macOS.

Comment (cransom) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:92]: i've got a build failure, https://gist.github.com/cransom/db7fe99aeec946ca216192a071059db0

Comment (Mic92) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:94]: I added libiconv. Does this fix the build for you?

**Comment by pinpox** [comment]
`20210404-112810-b63a949d` just was released yesterday, could we update here or should that be a separate PR?

**Comment by Mic92** [comment]
I close this for someone else to pick up. It still needs UserNotifications from frameworks on darwin as reported by @abathur: https://gist.github.com/abathur/775d43fdb28d9d9b833dfb2ad62b0d86

**Comment by SuperSandro2000** [comment]
We could also just update the linux portion and ignore darwin users for now.

**Comment by pinpox** [comment]
Since this PR is closed I continued the work over here: https://github.com/NixOS/nixpkgs/pull/118712

**Pull request closed without merging**